### ISSUE TITLE
failed to find loop device: could not attach image file to loop device: no loop devices available

### ISSUE THREAD

**Issue by brandon-biggs** [apptainer/apptainer#344]
We recently changed schedulers from PBS Pro to openpbs. After we switched, we noticed that when launching an interactive job, we can’t run singularity containers:
```
$ qsub -l select=1:ncpus=1:mpiprocs=1 -l walltime=03:00:00 -q general -I
$ singularity run alpine_latest.sif 
FATAL:   container creation failed: mount /proc/self/fd/10->/var/singularity/mnt/session/rootfs error: while mounting image /proc/self/fd/10: failed to find loop device: could not attach image file to loop device: no loop devices available
```
If I just ssh to the same compute node, it works fine or if I have an interactive job, and ssh to localhost or su to myself, it also works.
```
$ qsub -l select=1:ncpus=1:mpiprocs=1 -l walltime=03:00:00 -q general -I
$ ssh localhost
$ singularity run alpine_latest.sif 
Singularity> 
```

We’ve been looking through environment variables, openpbs settings, etc and can’t find a reason as to why this is happening. I was curious if anyone else had experienced this or if anyone had any thoughts. This works just fine on our Rocky 8 system that uses openpbs. This problem exists in two different versions of singularity and works fine without the scheduler.

### Version of Apptainer:

What version of Apptainer are you using? Run:

RPM -
```
$ singularity --version
singularity version 3.8.3-1.el7
```
Source -
```
$ singularity --version
singularity-ce version 3.9.0
```

### Expected behavior

What did you expect to see when you do...?

```
$ singularity run alpine_latest.sif 
Singularity> 
```

### Actual behavior

What actually happened? Why was it incorrect?
```
$ singularity run alpine_latest.sif 
FATAL:   container creation failed: mount /proc/self/fd/10->/var/singularity/mnt/session/rootfs error: while mounting image /proc/self/fd/10: failed to find loop device: could not attach image file to loop device: no loop devices available
```
For some reason, the container doesn't seem to run.

### Steps to reproduce this behavior

How can others reproduce this issue/problem?

I'm not sure how else to reproduce this issue.

I don't believe it's a lack of loop devices. There are 256 entries like this, 0-255
```
$ ls -l /dev/loop*
brw-rw---- 1 root disk  7,   0 Mar 22 13:47 /dev/loop0
brw-r----- 1 root disk  7,   1 Mar 22 13:03 /dev/loop1
brw-r----- 1 root disk  7,  10 Mar 22 13:03 /dev/loop10
brw-r----- 1 root disk  7, 100 Mar 22 13:03 /dev/loop100
...
crw-rw---- 1 root disk 10, 237 Mar 22 13:16 /dev/loop-control
```

### What OS/distro are you running

```
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

```

### How did you install Apptainer

Write here how you installed Apptainer. Eg. RPM, source.

Source and RPM

Additional output:

```
$ singularity -d run alpine_latest.sif
DEBUG   [U=41139,P=187871] persistentPreRun()            Singularity version: 3.8.3-1.el7
DEBUG   [U=41139,P=187871] persistentPreRun()            Parsing configuration file /etc/singularity/singularity.conf
DEBUG   [U=41139,P=187871] handleConfDir()               /home/[user_1]/.singularity already exists. Not creating.
DEBUG   [U=41139,P=187871] execStarter()                 Saving umask 0002 for propagation into container
DEBUG   [U=41139,P=187871] execStarter()                 Checking for encrypted system partition
DEBUG   [U=41139,P=187871] Init()                        Image format detection
DEBUG   [U=41139,P=187871] Init()                        Check for sandbox image format
DEBUG   [U=41139,P=187871] Init()                        sandbox format initializer returned: not a directory image
DEBUG   [U=41139,P=187871] Init()                        Check for sif image format
DEBUG   [U=41139,P=187871] Init()                        sif image format detected
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding __LMOD_REF_COUNT_GOPATH environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding I_MPI_HYDRA_BRANCH_COUNT environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding MANPATH environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding _ModuleTable003_ environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding HOSTNAME environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding __LMOD_REF_COUNT_MODULEPATH environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding TERM environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding SHELL environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_ROOT environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding HISTSIZE environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding MODULEPATH_ROOT environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_JOBNAME environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_PACKAGE_PATH environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_PKG environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_ENVIRONMENT environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding QTDIR environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding QTINC environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_SHORT_TIME environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_VERSION environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_O_WORKDIR environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding __LMOD_REF_COUNT_LOADEDMODULES environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding NCPUS environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding QT_GRAPHICSSYSTEM_CHECKED environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding HISTFILESIZE environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_TASKNUM environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding USER environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_sys environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LS_COLORS environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_O_HOME environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding __LMOD_REF_COUNT__LMFILES_ environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_MOMPORT environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_PREPEND_BLOCK environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_O_QUEUE environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding _ModuleTable001_ environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding MAIL environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_O_LOGNAME environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_O_LANG environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_JOBCOOKIE environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_SETTARG_CMD environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PWD environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding _LMFILES_ environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding CUDA_VISIBLE_DEVICES environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LANG environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_PAGER environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_NODENUM environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding MODULEPATH environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding _ModuleTable_Sz_ environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LOADEDMODULES environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_JOBDIR environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding TZ environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding C3_RSH environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_O_SHELL environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_CMD environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding HISTIGNORE environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_JOBID environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding SSH_ASKPASS environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding HISTCONTROL environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding SHLVL environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding __LMOD_REF_COUNT_PATH environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LANGUAGE environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding I_MPI_HYDRA_BOOTSTRAP_EXEC environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding _ModuleTable002_ environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_O_HOST environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding BASH_ENV environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_arch environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LOGNAME environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding QTLIB environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding XDG_DATA_DIRS environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_QUEUE environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding MODULESHOME environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding I_MPI_HYDRA_BOOTSTRAP environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_SETTARG_FULL_SUPPORT environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LESSOPEN environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding OMP_NUM_THREADS environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_O_MAIL environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PROMPT_COMMAND environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding USE_PCM_DB environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_FULL_SETTARG_SUPPORT environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding XDG_RUNTIME_DIR environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_DIR environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_O_SYSTEM environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding HISTTIMEFORMAT environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding LMOD_COLORIZE environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_NODEFILE environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding PBS_O_PATH environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding BASH_FUNC_module() environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding BASH_FUNC_ml() environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding _ environment variable
DEBUG   [U=41139,P=187871] SetContainerEnv()             Forwarding USER_PATH environment variable
VERBOSE [U=41139,P=187871] SetContainerEnv()             Setting HOME=/home/[user_1]
VERBOSE [U=41139,P=187871] SetContainerEnv()             Setting PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
DEBUG   [U=41139,P=187871] init()                        Use starter binary /usr/libexec/singularity/bin/starter-suid
VERBOSE [U=0,P=187871]     print()                       Set messagelevel to: 5
VERBOSE [U=0,P=187871]     init()                        Starter initialization
DEBUG   [U=0,P=187871]     load_overlay_module()         Trying to load overlay kernel module
DEBUG   [U=0,P=187871]     load_overlay_module()         Overlay seems supported by the kernel
VERBOSE [U=0,P=187871]     is_suid()                     Check if we are running as setuid
VERBOSE [U=0,P=187871]     priv_drop()                   Drop root privileges
DEBUG   [U=41139,P=187871]  read_engine_config()          Read engine configuration
DEBUG   [U=41139,P=187871]  init()                        Wait completion of stage1
VERBOSE [U=41139,P=187879]  priv_drop()                   Drop root privileges permanently
DEBUG   [U=41139,P=187879]  set_parent_death_signal()     Set parent death signal to 9
VERBOSE [U=41139,P=187879]  init()                        Spawn stage 1
DEBUG   [U=41139,P=187879] startup()                     singularity runtime engine selected
VERBOSE [U=41139,P=187879] startup()                     Execute stage 1
DEBUG   [U=41139,P=187879] StageOne()                    Entering stage 1
DEBUG   [U=41139,P=187879] prepareAutofs()               Found "/proc/sys/fs/binfmt_misc" as autofs mount point
DEBUG   [U=41139,P=187879] prepareAutofs()               Found "/misc" as autofs mount point
DEBUG   [U=41139,P=187879] prepareAutofs()               Found "/net" as autofs mount point
DEBUG   [U=41139,P=187879] prepareAutofs()               Found "/projects" as autofs mount point
DEBUG   [U=41139,P=187879] prepareAutofs()               Found "/hpc-common" as autofs mount point
DEBUG   [U=41139,P=187879] prepareAutofs()               Found "/home" as autofs mount point
DEBUG   [U=41139,P=187879] prepareAutofs()               Could not keep file descriptor for bind path /etc/localtime: no mount point
DEBUG   [U=41139,P=187879] prepareAutofs()               Could not keep file descriptor for bind path /etc/hosts: no mount point
DEBUG   [U=41139,P=187879] keepAutofsMount()             Open file descriptor for /home/[user_1]
DEBUG   [U=41139,P=187879] keepAutofsMount()             Open file descriptor for /home/[user_1]
DEBUG   [U=41139,P=187879] Init()                        Image format detection
DEBUG   [U=41139,P=187879] Init()                        Check for sandbox image format
DEBUG   [U=41139,P=187879] Init()                        sandbox format initializer returned: not a directory image
DEBUG   [U=41139,P=187879] Init()                        Check for sif image format
DEBUG   [U=41139,P=187879] Init()                        sif image format detected
DEBUG   [U=41139,P=187879] setSessionLayer()             Overlay seems supported and allowed by kernel
DEBUG   [U=41139,P=187879] setSessionLayer()             Attempting to use overlayfs (enable overlay = try)
VERBOSE [U=41139,P=187871]  wait_child()                  stage 1 exited with status 0
DEBUG   [U=41139,P=187871]  cleanup_fd()                  Close file descriptor 4
DEBUG   [U=41139,P=187871]  cleanup_fd()                  Close file descriptor 5
DEBUG   [U=41139,P=187871]  cleanup_fd()                  Close file descriptor 6
DEBUG   [U=41139,P=187871]  cleanup_fd()                  Close file descriptor 7
DEBUG   [U=41139,P=187871]  cleanup_fd()                  Close file descriptor 8
DEBUG   [U=41139,P=187871]  init()                        Set child signal mask
DEBUG   [U=41139,P=187871]  init()                        Create socketpair for master communication channel
DEBUG   [U=41139,P=187871]  init()                        Create RPC socketpair for communication between stage 2 and RPC server
VERBOSE [U=41139,P=187871]  priv_escalate()               Get root privileges
VERBOSE [U=0,P=187871]     priv_escalate()               Change filesystem uid to 41139
VERBOSE [U=0,P=187871]     init()                        Spawn master process
DEBUG   [U=0,P=187885]     set_parent_death_signal()     Set parent death signal to 9
VERBOSE [U=0,P=187885]     create_namespace()            Create mount namespace
VERBOSE [U=0,P=187871]     enter_namespace()             Entering in mount namespace
DEBUG   [U=0,P=187871]     enter_namespace()             Opening namespace file ns/mnt
DEBUG   [U=0,P=187871]     set_master_privileges()       Set master privileges
DEBUG   [U=0,P=187871]     apply_privileges()            Effective capabilities:   0x00000000000000c0
DEBUG   [U=0,P=187871]     apply_privileges()            Permitted capabilities:   0x0000001fffffffff
DEBUG   [U=0,P=187871]     apply_privileges()            Bounding capabilities:    0x0000001fffffffff
DEBUG   [U=0,P=187871]     apply_privileges()            Inheritable capabilities: 0x0000001fffffffff
DEBUG   [U=0,P=187871]     apply_privileges()            Ambient capabilities:     0x0000000000000000
DEBUG   [U=0,P=187871]     apply_privileges()            Set user ID to 41139
VERBOSE [U=0,P=187885]     create_namespace()            Create mount namespace
DEBUG   [U=0,P=187888]     set_rpc_privileges()          Set RPC privileges
DEBUG   [U=0,P=187888]     apply_privileges()            Effective capabilities:   0x0000000000200000
DEBUG   [U=0,P=187888]     apply_privileges()            Permitted capabilities:   0x0000001fffffffff
DEBUG   [U=0,P=187888]     apply_privileges()            Bounding capabilities:    0x0000000008204000
DEBUG   [U=0,P=187888]     apply_privileges()            Inheritable capabilities: 0x0000000000000000
DEBUG   [U=0,P=187888]     apply_privileges()            Ambient capabilities:     0x0000000000000000
DEBUG   [U=0,P=187888]     apply_privileges()            Set user ID to 41139
DEBUG   [U=41139,P=187888]  set_parent_death_signal()     Set parent death signal to 9
VERBOSE [U=41139,P=187888]  init()                        Spawn RPC server
DEBUG   [U=41139,P=187888] startup()                     singularity runtime engine selected
VERBOSE [U=41139,P=187888] startup()                     Serve RPC requests
DEBUG   [U=41139,P=187871] startup()                     singularity runtime engine selected
VERBOSE [U=41139,P=187871] startup()                     Execute master process
DEBUG   [U=41139,P=187871] setupSessionLayout()          Using Layer system: overlay
DEBUG   [U=41139,P=187871] setupOverlayLayout()          Creating overlay SESSIONDIR layout
DEBUG   [U=41139,P=187871] addRootfsMount()              Mount rootfs in read-only mode
DEBUG   [U=41139,P=187871] addRootfsMount()              Image type is 4096
DEBUG   [U=41139,P=187871] addRootfsMount()              Mounting block [squashfs] image: /home/[user_1]/alpine_latest.sif
DEBUG   [U=41139,P=187871] addKernelMount()              Checking configuration file for 'mount proc'
DEBUG   [U=41139,P=187871] addKernelMount()              Adding proc to mount list
VERBOSE [U=41139,P=187871] addKernelMount()              Default mount: /proc:/proc
DEBUG   [U=41139,P=187871] addKernelMount()              Checking configuration file for 'mount sys'
DEBUG   [U=41139,P=187871] addKernelMount()              Adding sysfs to mount list
VERBOSE [U=41139,P=187871] addKernelMount()              Default mount: /sys:/sys
DEBUG   [U=41139,P=187871] addDevMount()                 Checking configuration file for 'mount dev'
DEBUG   [U=41139,P=187871] addDevMount()                 Adding dev to mount list
VERBOSE [U=41139,P=187871] addDevMount()                 Default mount: /dev:/dev
DEBUG   [U=41139,P=187871] addHostMount()                Not mounting host file systems per configuration
VERBOSE [U=41139,P=187871] addBindsMount()               Found 'bind path' = /etc/localtime, /etc/localtime
VERBOSE [U=41139,P=187871] addBindsMount()               Found 'bind path' = /etc/hosts, /etc/hosts
DEBUG   [U=41139,P=187871] addHomeStagingDir()           Staging home directory (/home/[user_1]) at /var/singularity/mnt/session/home/[user_1]
DEBUG   [U=41139,P=187871] addHomeMount()                Adding home directory mount [/var/singularity/mnt/session/home/[user_1]:/home/[user_1]] to list using layer: overlay
DEBUG   [U=41139,P=187871] addTmpMount()                 Checking for 'mount tmp' in configuration file
VERBOSE [U=41139,P=187871] addTmpMount()                 Default mount: /tmp:/tmp
VERBOSE [U=41139,P=187871] addTmpMount()                 Default mount: /var/tmp:/var/tmp
DEBUG   [U=41139,P=187871] addScratchMount()             Not mounting scratch directory: Not requested
DEBUG   [U=41139,P=187871] addLibsMount()                Checking for 'user bind control' in configuration file
DEBUG   [U=41139,P=187871] addFilesMount()               Checking for 'user bind control' in configuration file
DEBUG   [U=41139,P=187871] addResolvConfMount()          Adding /etc/resolv.conf to mount list
VERBOSE [U=41139,P=187871] addResolvConfMount()          Default mount: /etc/resolv.conf:/etc/resolv.conf
DEBUG   [U=41139,P=187871] addHostnameMount()            Skipping hostname mount, not virtualizing UTS namespace on user request
DEBUG   [U=41139,P=187871] create()                      Mount all
DEBUG   [U=41139,P=187871] mountGeneric()                Mounting tmpfs to /var/singularity/mnt/session
FATAL   [U=41139,P=187871] Master()                      container creation failed: mount /proc/self/fd/10->/var/singularity/mnt/session/rootfs error: while mounting image /proc/self/fd/10: failed to find loop device: could not attach image file to loop device: no loop devices available
```

**Comment by DrDaveD** [comment]
To clarify, your "Source" version that works is on the Rocky 8 system, correct? There were quite some changes in loop device handling that went in to singularity-ce 3.9 and even more into apptainer-1.0.  It would be interesting to know if either of them fail on the same CentOS 7 environment where you're seeing this failure.

**Comment by brandon-biggs** [comment]
So we have two versions of singularity installed on the system that's failing. The first was installed via an RPM, the second was the source version. The source version is installed in our shared apps space and is not available unless you load our singularity module. It is the same version that is installed on our Rocky 8 system.

More details:
CentOS 7 system. Both of these fail
```
$ which singularity
/usr/bin/singularity
$ singularity --version
singularity version 3.8.3-1.el7
$ module load singularity
$ which singularity
/apps/singularity/3.9.0/bin/singularity
$ singularity --version
singularity-ce version 3.9.0
```

Rocky 8 system where it works
```
$ module load singularity
$ singularity --version
singularity-ce version 3.9.0
```

**Comment by DrDaveD** [comment]
An additional loop device fix went into apptainer 1.0.1.  Can you try it with apptainer-1.0.1?

**Comment by brandon-biggs** [comment]
Unfortunately (kind of) this resolved itself. We're not sure what exactly happened or why it's all of a sudden working, but neither issue seems to exist. We're expecting it to come back. If it does, we'll try apptainer-1.0.1. Thanks so much for the help!
